cadquery/assembly.py

```diff
@@ -543,6 +543,28 @@ def _flatten(self, parents=[]):
 
         return rv
 
+    def __iter__(
+        self,
+        loc: Optional[Location] = None,
+        name: Optional[str] = None,
+        color: Optional[Color] = None,
+    ) -> Iterator[Tuple[Shape, str, Location, Optional[Color]]]:
+        """
+        Assembly iterator yielding shapes, locatons, names and colors.
+        """
+
+        name = f"{name}/{self.name}" if name else self.name
+        loc = loc * self.loc if loc else self.loc
+        color = self.color if self.color else color
+
+        if self.obj:
+            yield self.obj if isinstance(self.obj, Shape) else Compound.makeCompound(
```

Comment:
The docstring order above is inconsistent with yield order here.

Change the order to (shape, location, name, color)  or change the docstring above to be consistent with the yield order?

Reply:
fixed